src/_response.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  CHALK_DATE_TYPES,
+  ChalkOnlineQueryRawData,
+  ChalkOnlineQueryRawResponse,
+} from "./_http";
+import { ChalkOnlineQueryResponse } from "./_interface";
+import { ChalkClientConfig, ChalkScalar } from "./_types";
+import { fromEntries } from "./_utils";
+import { mapRawResponseMeta } from "./_meta";
+
+type FeatureEntry<
+  TFeatureMap = Record<string, ChalkScalar>,
+  TOutput extends keyof TFeatureMap = keyof TFeatureMap
+> = ChalkOnlineQueryResponse<TFeatureMap, TOutput>["data"][any];
+
+const processDataValue = (
+  d: ChalkOnlineQueryRawData,
+  { timestampFormat }: Pick<ChalkClientConfig, "timestampFormat">
+): ChalkScalar => {
+  if (
+    timestampFormat === "EPOCH_MILLIS" &&
+    CHALK_DATE_TYPES.has(d.meta?.primitive_type)
+  ) {
+    return new Date(d.value).getTime();
+  }
+
+  return d.value;
+};
+
+export const parseOnlineQueryResponse = <
+  TFeatureMap = Record<string, ChalkScalar>,
+  TOutput extends keyof TFeatureMap = keyof TFeatureMap
+>(
+  rawResult: ChalkOnlineQueryRawResponse,
+  config: Pick<ChalkClientConfig, "timestampFormat">
+): ChalkOnlineQueryResponse<TFeatureMap, TOutput> => {
+  return {
+    data: fromEntries(
+      rawResult.data.map((d): [string, FeatureEntry] => {
+        return [
+          d.field,
+          {
+            value: processDataValue(d, config),
```

Comment:
It didn't occur to me that we might want to support EPOCH_MILLIS for regular queries too, and in that light I can see why you would want to default to ISO_8601 for backwards compatibility with _regular_ online query.